packages/core/core/src/BundleGraph.js

```diff
@@ -1146,23 +1144,40 @@ export default class BundleGraph {
       let referencedBundles = this.getReferencedBundles(bundle);
       for (let referenced of referencedBundles) {
         if (referenced.isInline) {
-          hash.update(this.getContentHash(referenced));
+          bundles.push(referenced);
           addReferencedBundles(referenced);
         }
       }
     };
 
     addReferencedBundles(bundle);
 
-    this.traverseBundles((childBundle, ctx, traversal) => {
-      if (childBundle.id === bundle.id || childBundle.isInline) {
-        hash.update(this.getContentHash(childBundle));
-      } else {
-        hash.update(childBundle.id);
+    this.traverseBundles((childBundle, _, traversal) => {
+      if (childBundle.isInline) {
+        bundles.push(childBundle);
+      } else if (childBundle.id !== bundle.id) {
         traversal.skipChildren();
       }
     }, bundle);
 
+    return bundles;
+  }
+
+  getHash(bundle: Bundle): string {
+    let hash = crypto.createHash('md5');
+    hash.update(bundle.id);
+    hash.update(this.getContentHash(bundle));
+
+    let inlineBundles = this.getInlineBundles(bundle);
+    for (let inlineBundle of inlineBundles) {
+      hash.update(this.getContentHash(inlineBundle));
+    }
+
+    this.traverseBundles((childBundle, _, traversal) => {
+      hash.update(childBundle.id);
+      traversal.skipChildren();
+    }, bundle);
```

Comment:
Does this do the same as before? Doesn't this traverse only the `childBundle` itself?

Reply:
I believe `bundleGraph.getChildBundles()` does exactly this

Reply:
Maybe?
```suggestion
    for(let childBundle of this.getChildBundles()) {
      hash.update(childBundle.id);
    }
```